hatch/src/cli/commands/build.rs

```diff
@@ -14,36 +16,35 @@ struct ImplicitBuilder;
 struct ExplicitBuilder;
 
 impl ImplicitBuilder {
-  fn get_project_names(&self, path: &String) -> Vec<OsString> {
-    let mut res_vec = Vec::new();
-
-    if let Ok(files) = fs::read_dir(path) {
-      for f in files {
-        if let Ok(unwrapped_f) = f {
-          let unwrapped_f = unwrapped_f.path();
-          if unwrapped_f.is_dir() {
-            if let Some(p) = unwrapped_f.file_stem() {
-              res_vec.push(p.to_os_string());
-            }
-          }
-        }
-      }
-    };
-
-    res_vec
+  fn get_project_names(&self, dir_paths: Vec<PathBuf>) -> Vec<String> {
+    dir_paths.iter()
+      .filter_map(|i| i.file_name())
+      .map(OsStr::new)
+      .filter_map(|i| i.to_str())
```

Comment:
Can we avoid the map to os string and just map directly to string?

Reply:
no -_-